x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_preview.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiEmptyPrompt, EuiFieldSearch, EuiPanel, EuiSpacer, EuiTitle } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { EngineLogic } from '../engine';
+import { Result } from '../result/result';
+
+import { RelevanceTuningLogic } from '.';
+
+const emptyCallout = (
+  <EuiEmptyPrompt
+    data-test-subj="EnterAQueryPrompt"
+    body={i18n.translate(
+      'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.preview.enterQueryMessage',
+      {
+        defaultMessage: 'Enter a query to see search results',
+      }
+    )}
+  />
+);
+
+const noResultsCallout = (
+  <EuiEmptyPrompt
+    data-test-subj="NoResultsPrompt"
+    body={i18n.translate(
+      'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.preview.noResultsMessage',
+      {
+        defaultMessage: 'No matching content found',
+      }
+    )}
+  />
+);
+
+export const RelevanceTuningPreview: React.FC = () => {
+  const { updateSearchValue } = useActions(RelevanceTuningLogic);
+  const { searchResults, schema } = useValues(RelevanceTuningLogic);
+  const { engineName, isMetaEngine } = useValues(EngineLogic);
+
+  return (
+    <EuiPanel>
+      <EuiTitle size="m">
+        <h2>
+          {i18n.translate('xpack.enterpriseSearch.appSearch.engine.relevanceTuning.preview.title', {
+            defaultMessage: 'Preview',
+          })}
+        </h2>
+      </EuiTitle>
+      <EuiSpacer />
+      <EuiFieldSearch
+        onChange={(e) => updateSearchValue(e.target.value)}
+        placeholder={i18n.translate(
+          'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.preview.textPlaceholder',
+          {
+            defaultMessage: 'Search {engineName}',
+            values: {
+              engineName,
+            },
+          }
+        )}
+        fullWidth
+      />
+      {!searchResults && emptyCallout}
+      {searchResults && searchResults.length === 0 && noResultsCallout}
+      {searchResults &&
+        searchResults.map((result) => {
+          return (
+            <React.Fragment key={result.id.raw}>
+              <EuiSpacer size="m" />
+              <Result
+                result={result}
+                showScore
+                isMetaEngine={isMetaEngine}
+                schemaForTypeHighlights={schema}
+              />
+            </React.Fragment>
+          );
+        })}
```

Comment:
you know I gotta do it... but what do you think... about a NESTED TERNARY????

```suggestion
      {searchResults
        ? searchResults.length
          ? searchResults.map((result) => (
              <React.Fragment key={result.id.raw}>
                <EuiSpacer size="m" />
                <Result
                  result={result}
                  showScore
                  isMetaEngine={isMetaEngine}
                  schemaForTypeHighlights={schema}
                />
              </React.Fragment>
            ))
          : noResultsCallout
        : emptyCallout}
```

lmfao mostly joking, I actually really prefer the way you wrote these branches. But something about me just _has_ to try writing nested ternaries just to see what it looks like :trollface:

Reply:
Lol, I don't hate it.